### DISCUSSION TITLE
Does this library use SSE/SIMD instructions for Matrix manipulation?

### DISCUSSION THREAD

**Discussion by AnReZa** [mathnet/mathnet-numerics#908] [Q&A]
Title says it all: Does this library use SSE/SIMD instructions for matrix manipulation, or is it purely based on traditional floating point arithmetic?

In my application, numerous looped matrix operations (multiplications and additions) take place. I wonder how useful it would be to switch to `System.Numerics.Matrix4x4`. Does anyone have any personal experience whether this would be beneficial?

**Comment by cdrnet** [comment]
The MKL native provider of Math.NET Numerics is leveraging SSE/AVX, but not yet the managed provider (the plan is to finally start with this on .NET 6 and newer after v5 is out).

However, if you're looking for high performance spatial/geometry code with lots of small 2-4-length vectors and matrices you're likely better of with the mentioned System.Numerics types at this point. Although there are plans to migrate Math.NET Spatial to these types as well, no one really had time to move forward with this.
